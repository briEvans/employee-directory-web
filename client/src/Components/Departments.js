import React, { Component } from 'react';

// Internal Routes
import '../App.scss';

class Departments extends Component {
  render() {
    return (
      <div className="app-body">
        <div className="container">
          <div className="header"><h2>Listing {this.props.category}</h2></div>
          <div className="employees">
            {
              this.props.departments.length > 0 ? (
                this.props.departments.map(item => {
                  return (
                    <div className="department">
                          <h3>{item}</h3>
                    </div>
                  );
                })
              ) : (<div className="employees"><h4> There are no Departments</h4></div>)
            }
          </div>
        </div>
      </div>
    );
  }
};

export default Departments;
