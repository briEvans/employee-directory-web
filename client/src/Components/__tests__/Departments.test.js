import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router, Route} from 'react-router-dom';
import { mount, shallow, render } from 'enzyme';

import Departments from '../Departments';

describe("Departments component", () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    <Router>
      ReactDOM.render(<Departments
        departments={['Department 1', 'Department2', 'Department3']}/>, div);
    </Router>
    ReactDOM.unmountComponentAtNode(div);
  });

  test("it matches the snapshot", () => {
    const component = shallow(<Departments
      departments={['Department 1', 'Department2', 'Department3']}/>);
    expect(component).toMatchSnapshot();
  });
});
