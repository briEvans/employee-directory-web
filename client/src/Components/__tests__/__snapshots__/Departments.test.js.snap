// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Departments component it matches the snapshot 1`] = `
<div
  className="app-body"
>
  <div
    className="container"
  >
    <div
      className="header"
    >
      <h2>
        Listing 
      </h2>
    </div>
    <div
      className="employees"
    >
      <div
        className="department"
        key="Department 1"
      >
        <h3>
          Department 1
        </h3>
      </div>
      <div
        className="department"
        key="Department2"
      >
        <h3>
          Department2
        </h3>
      </div>
      <div
        className="department"
        key="Department3"
      >
        <h3>
          Department3
        </h3>
      </div>
    </div>
  </div>
</div>
`;
